### PR TITLE
Update isSupportedDomain

### PR DESCRIPTION
**Pull request by JohnnyJumper**
Pivotal story:
https://www.pivotaltracker.com/story/show/178482911

The method should check the supported domain zone by making a network call to the ProxyReader (or proxy) (except .zil)

This PR relies on #45  and should be merged after.

### CODE REVIEW

**Review by rommex — COMMENTED** [review]
I guess my suggestions for the previous PR apply here as well

File: Sources/UnstoppableDomainsResolution/NamingServices/UNS.swift [hunk 1]
```diff
@@ -42,7 +42,20 @@ internal class CNS: CommonNamingService, NamingService {
...
+        let tld = split.suffix(1).joined(separator: "")
+        if tld == "zil" {
+            return false
+        }
+        let tokenId = self.namehash(domain: tld)
+        if let response = try? self.proxyReaderContract?.callMethod(methodName: Self.existName, args: [tokenId]) {
+            // swiftlint:disable force_cast
+            let result = response as! [String: Bool]
+            // swiftlint:enable force_cast
+            guard let isExist = result["0"] else { return false }
```
Comment (merenkoff) [inline Sources/UnstoppableDomainsResolution/NamingServices/UNS.swift:55]:
```suggestion
            guard 
              let result = response as? [String: Bool],
              let isExist = result["0"] else { 
                return false
            }
```

File: Sources/UnstoppableDomainsResolution/NamingServices/UNS.swift [hunk 2]
```diff
@@ -32,7 +32,7 @@ internal class CNS: CommonNamingService, NamingService {
             : config.network
 
         guard let contractsContainer = try Self.parseContractAddresses(network: network),
-              let registry = contractsContainer[ContractType.registry.name]?.address,
+              let registry = contractsContainer[ContractType.unsRegistry.name]?.address,
```
Comment (merenkoff) [inline Sources/UnstoppableDomainsResolution/NamingServices/UNS.swift:35]: How about old domains stored in the old CNS registry?

File: Sources/UnstoppableDomainsResolution/NamingServices/UNS.swift (hunk 1 above)
Comment (JohnnyJumper) [reply Sources/UnstoppableDomainsResolution/NamingServices/UNS.swift:55]: THANK YOU!!! I have tried to achieve exactly the same but couldn't make it work

**Comment by JohnnyJumper** [comment]
To make navigation on commits easier the base branch has been updated to #45, we will need to update it to uns once #45 is merged

**Review by DeRain — APPROVED** [review]
👍

**Review by merenkoff — APPROVED** [review]

**Pull request merged**